torch/jit/_recursive.py

```diff
@@ -68,9 +68,9 @@ def infer_raw_concrete_type(nn_module):
     """
     concrete_type = torch._C.ConcreteModuleType()
     concrete_type.add_pyclass(type(nn_module))
-    if isinstance(nn_module, (torch.nn.ModuleDict, torch.jit._ConstModuleDict)):
```

Comment:
I think this will break in a case like this, due to #29092

```python
class S(Sequential):
   ....

class M(Module):
    def __init__(self):
        super().__init__()
        self.x = torch.jit.script(S())

torch.jit.script(M())
```

Reply:
Looking into this example, however https://github.com/pytorch/pytorch/issues/29092 should be fixed on its own right as well.

Reply:
This works:
```
        class Inner(torch.nn.Module):
            def forward(self, x):
                return x + 10

        class CustomSequential(nn.Sequential):
            def __init__(self):
                super(CustomSequential, self).__init__(
                    nn.ReLU(), Inner())

            def forward(self, x):
                x = x + 3
                for mod in self:
                    x = mod(x)
                return x - 5

        class M(nn.Module):
            def __init__(self):
                super().__init__()
                self.x = torch.jit.script(CustomSequential())

            def forward(self, input):
                return self.x.forward(input)

        self.checkModule(M(), (torch.tensor(.5),))
```